dpu-cni/pkgs/cniserver/cniserver.go

```diff
@@ -24,17 +25,19 @@ import (
 // This server implementations is only temporary for testing when the DPU Daemon
 // code has not been implemented.
 
-type processRequestFunc func(request *cnitypes.Request) (*cni100.Result, error)
+type processRequestFunc func(request *cnitypes.PodRequest) (*cni100.Result, error)
 type Server struct {
 	http.Server
-	processRequest processRequestFunc
-	runDir         string
-	socketPath     string
+	addHandler processRequestFunc
+	delHandler processRequestFunc
+	socketPath string
 }
 
 // ensureRunDirExists makes sure that the socket being created is only accessible to root.
-func ensureRunDirExists(runDir string, socketPath string) error {
+func ensureRunDirExists(serverSocketPath string) error {
```

Comment:
why is this needed?
on deletion should the socket not get cleaned up?

Reply:
No, we must ensure that we are dealing with a clean slate. You can't assume anything about the state of the filesystem.